### PR TITLE
scootapi/scoot - Add timeouts to scootapi thrift, support in client run_job

### PR DESCRIPTION
**Pull request by dgassaway**
https://jira.twitter.biz/browse/DX-2130
* add timeouts to job/task def in scootapi
* job def timeout applies to tasks, unless specified in task def
* add equivalent fields to client run_job structs
* fix 2 `go vet` unspecified field errors

### CODE REVIEW

File: scootapi/client/run_job_cmd.go [hunk 1]
```diff
@@ -33,13 +33,15 @@ func (c *runJobCmd) registerFlags() *cobra.Command {
 
 // Types to handle JobDefinitions from JSON files
 type JobDef struct {
-	Tasks map[string]TaskDef
+	Tasks     map[string]TaskDef
+	TimeoutMs int32
```
Comment (jasonbs10) [inline scootapi/client/run_job_cmd.go]: We should take this out until we're ready to implement it.

**Comment by dgassaway** [comment]
I think this might actually be covered by scootapi/server/translate_prop_test.go, but I also manually verified it

File: scootapi/client/run_job_cmd.go (hunk 1 above)
Comment (dgassaway) [inline scootapi/client/run_job_cmd.go]: Not following this

Comment (jasonbs10) [inline scootapi/client/run_job_cmd.go]: Task timeout appears to flow through but job timeout is dropped in scheduler. We probably shouldn't add things unless we have a concrete timeline for using them.

Comment (dgassaway) [inline scootapi/client/run_job_cmd.go]: Ah gotcha - these get resolved in scootapi/server/run_job.go - task timeout will be used, if it's not set then job timeout will be used

Comment (jasonbs10) [inline scootapi/client/run_job_cmd.go]: Does that conform to expected behavior? Job timeout should either be called defaultTaskTimeout in this case or it should be used to limit total job time imo.

Comment (dgassaway) [inline scootapi/client/run_job_cmd.go]: Agreed on defaultTaskTimeout - it's just a shorthand vs including the field in each task. Limiting job time would be trickier, and different use case.

**Pull request merged**